File: common_base/src/main/java/com/yyxnb/common_base/weight/skin/RecycleViewResDeployer.java
package com.yyxnb.common_base.weight.skin;

import android.support.v7.widget.RecyclerView;
import android.view.View;

import com.yyxnb.lib_adapter.ItemDecoration;
import com.yyxnb.skinloader.bean.SkinAttr;
import com.yyxnb.skinloader.bean.SkinConfig;
import com.yyxnb.skinloader.skinInterface.ISkinResDeployer;
import com.yyxnb.skinloader.skinInterface.ISkinResourceManager;


public class RecycleViewResDeployer implements ISkinResDeployer {

    private int divHeight;

    public RecycleViewResDeployer() {
    }

    public RecycleViewResDeployer(int divHeight) {
        this.divHeight = divHeight;
    }

    @Override
    public void deploy(View view, SkinAttr skinAttr, ISkinResourceManager resource) {
        if (!(view instanceof RecyclerView)) {
            return;
        }

        RecyclerView recyclerView = (RecyclerView) view;
        if (SkinConfig.RES_TYPE_NAME_COLOR.equals(skinAttr.attrValueTypeName)) {
            ItemDecoration decoration = new ItemDecoration(recyclerView.getContext());
            if (divHeight > 0) {
                decoration.setDividerHeight(divHeight);
            }
            decoration.setDividerColor(resource.getColor(skinAttr.attrValueRefId));

            recyclerView.removeItemDecorationAt(0);
            recyclerView.addItemDecoration(decoration);

//            LogUtils.e("-----" + skinAttr.attrValueTypeName + " , " + skinAttr.attrValueRefId);
        }


    }
}

File: common_base/src/main/java/com/yyxnb/common_base/weight/skin/RecycleViewResDeployer.kt
package com.yyxnb.common_base.weight.skin

import android.view.View
import androidx.recyclerview.widget.RecyclerView
import com.yyxnb.adapter.ItemDecoration
import com.yyxnb.skinloader.bean.SkinAttr
import com.yyxnb.skinloader.bean.SkinConfig
import com.yyxnb.skinloader.skinInterface.ISkinResDeployer
import com.yyxnb.skinloader.skinInterface.ISkinResourceManager

class RecycleViewResDeployer : ISkinResDeployer {
    private var divHeight = 0

    constructor() {}
    constructor(divHeight: Int) {
        this.divHeight = divHeight
    }

    override fun deploy(view: View, skinAttr: SkinAttr, resource: ISkinResourceManager) {
        if (view !is RecyclerView) {
            return
        }
        val recyclerView = view
        if (SkinConfig.RES_TYPE_NAME_COLOR == skinAttr.attrValueTypeName) {
            val decoration = ItemDecoration(recyclerView.context)
            if (divHeight > 0) {
                decoration.setDividerHeight(divHeight)
            }
            decoration.setDividerColor(resource.getColor(skinAttr.attrValueRefId))
            recyclerView.removeItemDecorationAt(0)
            recyclerView.addItemDecoration(decoration)

//            LogUtils.e("-----" + skinAttr.attrValueTypeName + " , " + skinAttr.attrValueRefId);
        }
    }
}